gnd/src/main/java/com/google/android/gnd/rx/Loadable.java

```diff
@@ -106,4 +102,29 @@ public String toString() {
       return state.toString();
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof Loadable)) {
+      return false;
+    }
+    Loadable<?> loadable = (Loadable<?>) o;
+    return getState() == loadable.getState();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(getState());
+  }
```

Comment:
What are equals() and hasCode() ever used? If not, please remove.

Reply:
These are used in the unit tests